core/src/Streamly/Internal/Data/Builder.hs

```diff
@@ -27,16 +27,16 @@ import Control.Applicative (liftA2)
 -- or even a Fold. Unlike fold the step function is one-shot and not called in
 -- a loop.
 newtype Builder s m a =
-  Builder (s -> m (s, a))
+  Builder (s -> m (a, s))
 
 -- | Maps a function on the output of the fold (the type @b@).
 instance Functor m => Functor (Builder s m) where
     {-# INLINE fmap #-}
-    fmap f (Builder step1) = Builder (fmap (fmap f) . step1)
+    fmap f (Builder step1) = Builder (fmap (\ ~(a, s') -> (f a, s')) . step1)
```

Comment:
@adithyaov this is how we have to map if we have value as the first parameter. Should we keep API divergence from StateT for mapping convenience?

Reply:
What is the reason for diverging in the first place? To conform with `StateT` should not be a reason.